main.py

```diff
@@ -152,7 +153,10 @@ def get_candidate_solution(self):
     def run_esp(self):
         """ Runs ESP, with an optional presolve with SDP """
 
-        logger.info(f"Running ESP in mode {self.mode} with {len(self.esp.shifts_set['shifts'])}")
+        if self.mode != "implicit":
```

Comment:
Should be 
```python
if self.mode != "implicit" and self.mode != 3
```

Reply:
or i stedet for and, eller?

Reply:
Nope, this is the correct way. You want this code to run if you have chosen  mode that is not _implicit_ / 3.

|  Input/Result | AND | OR |
| ------------- | ------------- | ------------- |
| _implicit_  | `False` due to first term | `True` due to second term  |
| 3 | `False` due to second term  | `True` due to first term |
| Anything else  | `True`  | `True`  |